leakcanary-object-watcher/src/main/java/leakcanary/Clock.kt

```diff
@@ -3,27 +3,17 @@ package leakcanary
 /**
  * An interface to abstract the SystemClock.uptimeMillis() Android API in non Android artifacts.
  *
- * You can create a [Clock] from a lambda by calling [invoke].
+ * This is a functional interface with which you can create a [Clock] from a lambda.
  */
-interface Clock {
+fun interface Clock {
   /**
    * On Android VMs, this should return android.os.SystemClock.uptimeMillis().
    */
   fun uptimeMillis(): Long
 
   companion object {
-    /**
-     * Utility function to create a [Clock] from the passed in [block] lambda
-     * instead of using the anonymous `object : Clock` syntax.
-     *
-     * Usage:
-     *
-     * ```kotlin
-     * val clock = Clock {
-     *
-     * }
-     * ```
-     */
+
+    @Deprecated("Leverage Kotlin SAM lambda expression")
```

Comment:
We probably shouldn't deprecate this because anyone using 1.3 still have to use this. Is there a way to deprecate only for Kotlin 1.4?

Reply:
Surprisingly Android Studio 4.1 does not strikethrough the usage when the client is using the invoke hack, whether on 1.3 or 1.4. So this deprecated annotation is purely informational.

In addition, we should set a timeline for us to clean up this hack. It could be when Kotlin 1.5 is officially released, so that we set our language compatibility promises to be only one major version below the current stable release.

Reply:
Generally we've shied away from dropping compatibility support unless there's a really good reason (ie it's blocking some other work). For example, when Android X happened and created a huge migration mess, we ended up removing the direct dependency on both support lib & android X so that we wouldn't have to pick one or force people to go through a painful migration until they were ready for it. Another example is Okio where we depend on a really old version so as not to force anyone to upgrade. Generally I don't want LeakCanary's adoption to be blocked because it's forcing someone to upgrade :) .

Reply:
Reverted the deprecation.